@here/harp-mapview/test/SphereHorizonTest.ts

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { GeoCoordinates, sphereProjection } from "@here/harp-geoutils";
+import { expect } from "chai";
+import { MathUtils, PerspectiveCamera, Vector3 } from "three";
+
+import { CanvasSide, SphereHorizon } from "../lib/SphereHorizon";
+import { MapViewUtils } from "../lib/Utils";
+
+describe("SphereHorizon", function() {
+    const eps = 1e-10;
+    const vFov = 40;
+    const canvasHeight = 800;
+    const projection = sphereProjection;
+    let focalLength = 256;
+    let camera: PerspectiveCamera;
+    const camXAxis = new Vector3();
+    const camYAxis = new Vector3();
+    const camZAxis = new Vector3();
+    let horizon: SphereHorizon;
+
+    function setCamera(
+        zoomLevel: number,
+        tilt: number = 0,
+        heading: number = 0,
+        geoTarget: GeoCoordinates = new GeoCoordinates(0, 0)
+    ) {
+        MapViewUtils.getCameraRotationAtTarget(
+            projection,
+            geoTarget,
+            -heading,
+            tilt,
+            camera.quaternion
+        );
+        const distance = MapViewUtils.calculateDistanceFromZoomLevel({ focalLength }, zoomLevel);
+        MapViewUtils.getCameraPositionFromTargetCoordinates(
+            geoTarget,
+            distance,
+            -heading,
+            tilt,
+            projection,
+            camera.position
+        );
+        camera.updateMatrixWorld(true);
+        camera.matrix.extractBasis(camXAxis, camYAxis, camZAxis);
+        horizon = new SphereHorizon(camera);
+    }
+
+    function checkPointInHorizon(point: Vector3) {
+        const tangent = new Vector3().subVectors(camera.position, point);
+        expect(Math.abs(tangent.angleTo(point))).closeTo(Math.PI / 2, eps);
+    }
+
+    function getHorizonPointT(point: Vector3) {
+        checkPointInHorizon(point);
+        const pointVec = point.clone().normalize();
+        const x = pointVec.dot(camXAxis);
+        const y = pointVec.dot(camYAxis);
+        const angle = Math.atan2(y, x);
+        return angle >= 0 ? angle / (2 * Math.PI) : 1 + angle / (2 * Math.PI);
+    }
+
+    before(function() {
+        focalLength = MapViewUtils.calculateFocalLengthByVerticalFov(
```

Comment:
MINOR: vFov and canvasHeight are both const, why not just calculate this once?

Reply:
It is calculated once for the whole test suite. The "before" function is run once before any test in the suite is run.

Reply:
Sorry, I mistook for beforeEach :(